cmd/assert.go

```diff
@@ -64,81 +64,92 @@ var assertCmd = &cobra.Command{
 	`,
 	Run: func(cmd *cobra.Command, args []string) {
 		// build experiment
-		exp := &experiment{
+		exp := &Experiment{
 			Experiment: &base.Experiment{},
 		}
 		log.Logger.Trace("build started")
-		exp, err := build(true)
+		// replace FileExpIO with ClusterExpIO to build from cluster
+		fio := &FileExpIO{}
+		exp, err := Build(true, fio)
 		log.Logger.Trace("build finished")
 		if err != nil {
 			log.Logger.Error("experiment build failed")
 			os.Exit(1)
 		}
 
-		// check assert conditions
-		allGood := true
-		for {
-			for _, cond := range conds {
-				if strings.ToLower(cond) == completed {
-					c := exp.completed()
-					allGood = allGood && c
-					if c {
-						log.Logger.Info("experiment completed")
-					} else {
-						log.Logger.Info("experiment did not complete")
-					}
-				} else if strings.ToLower(cond) == noFailure {
-					nf := exp.noFailure()
-					allGood = allGood && nf
-					if nf {
-						log.Logger.Info("experiment has no failure")
-					} else {
-						log.Logger.Info("experiment failed")
-					}
-				} else if strings.ToLower(cond) == slos {
-					slos := exp.SLOs()
-					allGood = allGood && slos
-					if slos {
-						log.Logger.Info("SLOs are satisfied")
-					} else {
-						log.Logger.Info("SLOs are not satisfied")
-					}
-				} else if strings.HasPrefix(cond, slosByPrefix) {
-					version, err := exp.extractVersion(cond)
-					if err != nil {
-						os.Exit(1)
-					}
-					iv := exp.SLOsBy(version)
-					allGood = allGood && iv
-					if iv {
-						log.Logger.Info(version, " satisfies objectives")
-					} else {
-						log.Logger.Info(version, " does not satisfy objectives")
-					}
+		allGood, err := exp.Assert(conds, timeout)
+		if err != nil || !allGood {
+			os.Exit(1)
+		}
+	},
+}
+
+// Assert if experiment satisfies conditions
+func (exp *Experiment) Assert(conditions []string, to time.Duration) (bool, error) {
+	// check assert conditions
+	allGood := true
+	for {
+		for _, cond := range conditions {
+			if strings.ToLower(cond) == completed {
```

Comment:
Could the constants `completed`, `noFailure`. `slos` and `slosByPrefix` be exposed? I create them as well...